app/controllers/assignments_controller.rb

```diff
@@ -28,26 +30,49 @@ def create
   end
 
   # rubocop:disable MethodLength
-  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable AbcSize
   def show
     @assignment_repos = AssignmentRepo
       .where(assignment: @assignment)
       .order(:id)
       .page(params[:page])
+
     return unless @organization.roster
 
     @roster_entries = @organization.roster.roster_entries
-      .order(:id)
       .page(params[:students_page])
       .order_for_view(@assignment)
+      .order_by_sort_mode(@current_sort_mode, assignment: @assignment)
+      .order(:id)
 
     @unlinked_user_repos = AssignmentRepo
       .where(assignment: @assignment, user: @unlinked_users)
       .order(:id)
       .page(params[:unlinked_accounts_page])
   end
+  # rubocop:enable AbcSize
+  # rubocop:enable MethodLength
+
+  # rubocop:disable MethodLength
+  def filter_repos
```

Comment:
As we discussed in Slack, lets :fire: this endpoint if possible. Instead of setting page(1) in the controller, that can happen in the front end on search.